### ISSUE TITLE
Strange behaviour of `sine :: ControlPattern`

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/Tidal#678]
```
segment 2 sine :: Pattern ControlMap
```
Gives:
```
(0>½)|n: 2, speed: 1.0f
(½>1)|n: 2, speed: -1.0f
```
Two strange things:

1. `sine` normally goes from `0` to `1`. Here it goes from `-1` to `1`. How?
2. Where does the `n 2` come from?

This is borderline paranormal.

**Comment by jwaldmann** [comment]
Wat? Can confirm! - With Tidal 1.0.0, I get 
```
segment 2 sine :: Pattern ControlMap 
(0>½)|n: 2, speed: 0.0f
(½>1)|n: 2, speed: 1.2246467991473532e-16f
```

**Comment by jwaldmann** [comment]
```
:set -ddump-simpl

segment 2 sine :: Pattern ControlMap 

==================== Simplified expression ====================
let {
  it_aDXu
    :: Sound.Tidal.Pattern.Pattern Sound.Tidal.Pattern.ControlMap
  [LclId,
   Unf=Unf{Src=<vanilla>, TopLvl=False, Value=False, ConLike=False,
           WorkFree=False, Expandable=False, Guidance=IF_ARGS [] 220 0}]
  it_aDXu
    = Sound.Tidal.UI.segment
        @ Sound.Tidal.Pattern.ControlMap
        (GHC.Num.fromInteger
           @ (Sound.Tidal.Pattern.Pattern Sound.Tidal.Pattern.Time)
           (Sound.Tidal.Pattern.$fNumPattern
              @ (GHC.Real.Ratio GHC.Integer.Type.Integer)
              (GHC.Real.$fNumRatio
                 @ GHC.Integer.Type.Integer GHC.Real.$fIntegralInteger))
           2)
        (Sound.Tidal.Core.sine
           @ Sound.Tidal.Pattern.ControlMap
           Sound.Tidal.Pattern.$fFractionalMap) } in
GHC.Base.thenIO
...
```

**Comment by bgold-cosmos** [comment]
It seems in this case the fundamental cause is that ControlMap has been declared an instance of Num, which is why something like `sine :: Pattern ControlMap` is allowed in the first place.

**Comment by yaxu** [comment]
Yes I originally made this feature for 8 year olds so they could make music just by doing `"0 1 2" # "clap"`. It found its way into the main release kind of by accident. It just seems to cause confusion.

Thanks for tracking this down @jwaldmann, it just made no sense to me, especially as `square` etc didn't show the same behaviour. But the simple case of `segment 2 $ sig (\t -> 2 + fromRational t) :: ControlPattern` clears it up.

I'm not sure if there's anything to fix here beyond #679, so closing for now.

**Comment by yaxu** [comment]
Or maybe we should not declare ControlMap as being in Num? That has caused extra confusion here, and probably isn't useful on its own.. Although it's a logical step in declaring `Num a => Pattern a` into Num, which despite being a bit confusing, is very useful.
